arxiv/auth/openid/tests/test_keycloak.py

```diff
@@ -0,0 +1,44 @@
+import subprocess
+import pytest
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+import time
+
+@pytest.fixture(scope="module")
+def web_driver() -> webdriver.Chrome:
+    # Set up the Selenium WebDriver
+    _web_driver = webdriver.Chrome()  # Ensure you have ChromeDriver installed and in PATH
+    _web_driver.implicitly_wait(10)  # Wait for elements to be ready
+    yield _web_driver
+    _web_driver.quit()  # Close the browser window after tests
+
+@pytest.fixture(scope="module")
+def toy_flask():
```

Comment:
Unless there is some specific reason to run flask in a sub process, consider running it as described here https://flask.palletsprojects.com/en/3.0.x/testing/

In the past we've written tests like this to use a client with cookies, get the form, post the form with login and then get a protected page. 

Now I'm seeing that the web_driver is requesting to flask running in the subprocess. This actually seems fine.

Reply:
Keycloak (KC) can run locally as a fixture, once what I want to test is more concrete. I needed to see what comes back from KC while changing the settings of KC - such as KC's notion of groups, roles, attributes you can define to the user.  For example, I haven't been able to get the user's group in the oauth2 reply.   

To make this a fixture, I need to run KC docker, and run a set of REST API calls to set up the settings (create the realm, client ID, client callback, etc.) 

This PR is more for not losing work and exploring what I can do with KC.  Right now, this is a test bench, and at some point, I will turn this into real regression test.